src/types/tdigest.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#pragma once
+
+#include <fmt/format.h>
+
+#include <vector>
+
+#include "status.h"
+
+struct Centroid {
+  double mean;
+  double weight = 1.0;
+
+  // merge with another centroid
+  void Merge(const Centroid& centroid) {
+    weight += centroid.weight;
+    mean += (centroid.mean - mean) * centroid.weight / weight;
+  }
+
+  std::string ToString() const { return fmt::format("centroid<mean: {}, weight: {}>", mean, weight); }
+
+  template <typename T>
+  explicit Centroid(T&& centroid) : mean(centroid.mean), weight(centroid.weight) {}
```

Comment:
Why not just leave a copy ctor here?

Reply:
Hi @mapleFU ,

Thanks for your suggestion! 😄
I have updated it to a default copy ctor.

Best Regards,
Edward